### PR TITLE
Homework 18

### PR DESCRIPTION
**Pull request by litserkovna**

### CODE REVIEW

File: tests/homework18/homework18.spec.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
+import { test, expect } from '@playwright/test';
+
+test.describe("Registration Form Tests", () => {
+
+  test.beforeEach(async ({ page }) => {
+    await page.goto("");
+    console.log('Page URL:', page.url());
+    const SignUpButton = page.locator('.hero-descriptor_btn.btn.btn-primary');
+    await SignUpButton.click();
+  })
```
Comment (sttaran) [inline tests/homework18/homework18.spec.js]: variable name should not start with upper letter

File: tests/homework18/homework18.spec.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+  test.describe('Positive Scenario', () => { 
+    test.afterEach('Removing created user', async ({ page }) => {
+      const SettingsButton = page.locator('a.btn.btn-white.btn-sidebar.sidebar_btn:has-text("Settings")');
+      await SettingsButton.click();
+      await expect(page).toHaveURL("/panel/settings");
+      await page.click('button.btn.btn-danger-bg:has-text("Remove my account")');
+      await page.click('button.btn.btn-danger:has-text("Remove")');
```
Comment (sttaran) [inline tests/homework18/homework18.spec.js]: why is it so complicated?

you can use hasText
example
` page.locator('x-details', { hasText: 'Details' })`

File: tests/homework18/homework18.spec.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  const emailInputErrorMessage = page.locator('div.invalid-feedback p');
+  await emailInput.fill('test.email');
+  await emailInput.focus();
+  await emailInput.blur();
+  await expect(emailInputErrorMessage).toHaveText('Email is incorrect');
+});
+//Password Field
+ test('Empty Password field validation', async ({ page }) => {
+  const passwordInput = page.locator('input[name="password"]');
+  const passwordError = page.locator('div.invalid-feedback p');
```
Comment (sttaran) [inline tests/homework18/homework18.spec.js]: `const passwordError = passwordInput.locator('+ div.invalid-feedback p');`

Comment (sttaran) [reply tests/homework18/homework18.spec.js]: locate by relying on input, same for other error locators

File: tests/homework18/homework18.spec.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  await expect(passwordError).toHaveText('Password required');
+  });
+
+  test('Invalid Password data - Incorrect format', async ({ page }) => {
+    const passwordInput = page.locator('input[name="password"]');
+    const passwordError = page.locator('div.invalid-feedback p');
+    await passwordInput.fill('test.email',{ timeout: 2000 });
+    await passwordInput.focus();
+    await passwordInput.blur();
+    await expect(passwordError).toHaveText('Password has to be from 8 to 15 characters long and contain at least one integer, one capital, and one small letter');  });
```
Comment (sttaran) [inline tests/homework18/homework18.spec.js]: check input border color

File: tests/homework18/homework18.spec.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+  await repeatPasswordInput.fill('');
+  await repeatPasswordInput.focus();
+  await repeatPasswordInput.blur();
+  await expect(secondPasswordError).toHaveText('Re-enter password required');
+});
+
+test('Invalid Re-enter Password data - Passwords do not match', async ({ page }) => {
+  const repeatPasswordInput = page.locator('input[name="repeatPassword"]');
+  const secondPasswordError = page.locator('div.invalid-feedback p');
+  await repeatPasswordInput.fill('passWord1');
```
Comment (sttaran) [inline tests/homework18/homework18.spec.js]: do not hardcode values

**Review by sttaran — CHANGES_REQUESTED** [review]

File: tests/homework18/homework18.spec.js (hunk 4 above)
Comment (litserkovna) [reply tests/homework18/homework18.spec.js]: await expect(nameErrorMessage).toHaveCSS('border-color', 'rgb(220, 53, 69)');

File: tests/homework18/homework18.spec.js (hunk 5 above)
Comment (litserkovna) [reply tests/homework18/homework18.spec.js]: const newPassword = 'passWord1';
  await repeatPasswordInput.fill(newPassword);

File: tests/homework18/homework18.spec.js (hunk 3 above)
Comment (litserkovna) [reply tests/homework18/homework18.spec.js]: Updated everywhere

File: tests/homework18/homework18.spec.js (hunk 2 above)
Comment (litserkovna) [reply tests/homework18/homework18.spec.js]: Updated

File: tests/homework18/homework18.spec.js (hunk 1 above)
Comment (litserkovna) [reply tests/homework18/homework18.spec.js]: Corrected

File: tests/homework18/homework18.spec.js (hunk 4 above)
Comment (sttaran) [reply tests/homework18/homework18.spec.js]: why do you check border color only for name input? apply this check for other inputs too

**Pull request merged**